File: src/types/import.ts
// File: src/types/import.ts

export interface CsvPreview {
  headers: string[];
  rows: string[][];
  total_rows: number;
  detected_delimiter: string;
}

export interface ColumnMapping {
  date_col: number;
  amount_col: number;
  type_col: number | null;
  account_col: number | null;
  category_col: number | null;
  memo_col: number | null;
  date_format: string;
  negative_as_expense: boolean;
}

export interface RowValidation {
  row_index: number;
  status: "valid" | "warning" | "error";
  date: string;
  amount: number;
  transaction_type: string;
  account_name: string;
  category_name: string;
  memo: string;
  error: string | null;
  matched_account_id: number | null;
  matched_category_id: number | null;
}

export interface ImportValidationResult {
  valid_count: number;
  warning_count: number;
  error_count: number;
  rows: RowValidation[];
  unmatched_accounts: string[];
  unmatched_categories: string[];
}

export interface ImportOptions {
  skip_duplicates: boolean;
  create_missing_categories: boolean;
  default_account_id: number;
  account_mapping: Record<string, number>;
  category_mapping: Record<string, number>;
}

export interface ImportResult {
  batch_id: string;
  imported: number;
  skipped: number;
  errors: number;
}

export interface ImportHistoryEntry {
  id: number;
  batch_id: string;
  filename: string;
  total_rows: number;
  imported_count: number;
  skipped_count: number;
  error_count: number;
  status: string;
  imported_at: string;
  can_undo: boolean;
}

export interface MatchSuggestion {
  name: string;
  matched_id: number | null;
  matched_name: string | null;
  score: number;
}

File: src-tauri/src/models/import.rs
// File: src-tauri/src/models/import.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Serialize)]
pub struct CsvPreview {
    pub headers: Vec<String>,
    pub rows: Vec<Vec<String>>,
    pub total_rows: i64,
    pub detected_delimiter: String,
}

#[derive(Debug, Deserialize)]
pub struct ColumnMapping {
    pub date_col: usize,
    pub amount_col: usize,
    pub type_col: Option<usize>,
    pub account_col: Option<usize>,
    pub category_col: Option<usize>,
    pub memo_col: Option<usize>,
    pub date_format: String,
    pub negative_as_expense: bool,
}

#[derive(Debug, Serialize)]
pub struct RowValidation {
    pub row_index: i64,
    pub status: String, // "valid", "warning", "error"
    pub date: String,
    pub amount: f64,
    pub transaction_type: String,
    pub account_name: String,
    pub category_name: String,
    pub memo: String,
    pub error: Option<String>,
    pub matched_account_id: Option<i64>,
    pub matched_category_id: Option<i64>,
}

#[derive(Debug, Serialize)]
pub struct ImportValidationResult {
    pub valid_count: i64,
    pub warning_count: i64,
    pub error_count: i64,
    pub rows: Vec<RowValidation>,
    pub unmatched_accounts: Vec<String>,
    pub unmatched_categories: Vec<String>,
}

#[derive(Debug, Deserialize)]
pub struct ImportOptions {
    pub skip_duplicates: bool,
    pub create_missing_categories: bool,
    pub default_account_id: i64,
    pub account_mapping: HashMap<String, i64>,
    pub category_mapping: HashMap<String, i64>,
}

#[derive(Debug, Serialize)]
pub struct ImportResult {
    pub batch_id: String,
    pub imported: i64,
    pub skipped: i64,
    pub errors: i64,
}

#[derive(Debug, Serialize)]
pub struct ImportHistoryEntry {
    pub id: i64,
    pub batch_id: String,
    pub filename: String,
    pub total_rows: i64,
    pub imported_count: i64,
    pub skipped_count: i64,
    pub error_count: i64,
    pub status: String,
    pub imported_at: String,
    pub can_undo: bool,
}

#[derive(Debug, Serialize)]
pub struct MatchSuggestion {
    pub name: String,
    pub matched_id: Option<i64>,
    pub matched_name: Option<String>,
    pub score: f64, // 0.0 to 1.0
}
